nav2_rotation_shim_controller/src/nav2_rotation_shim_controller.cpp

```diff
@@ -139,6 +144,42 @@ geometry_msgs::msg::TwistStamped RotationShimController::computeVelocityCommands
   const geometry_msgs::msg::Twist & velocity,
   nav2_core::GoalChecker * goal_checker)
 {
+  // Rotate to goal heading when in goal xy tolerance
+  if (rotate_to_goal_heading_) {
+    std::lock_guard<std::mutex> lock_reinit(mutex_);
+
+    try {
+      geometry_msgs::msg::Pose base_pose = transformPoseToBaseFrame(pose);
```

Comment:
Why transfer the pose into base frame? Can't you use this as is and only transform the path's final pose (goal) to the `pose.header.frame_id` frame?